### PR TITLE
Tdl 23543 fix uuid transformation

### PR DESCRIPTION
**Pull request by RushiT0122**
# Description of change

- Fix the uuid transformation issue because UUIDLegacy support was removed by [`PyMongo v4.3+`](https://pymongo.readthedocs.io/en/stable/migrate-to-pymongo4.html#uuidlegacy-is-removed) and reads UUID fields as `bson.binary.Binary` object causes the transform issue.
- Fix multiple compatibility issues in integration tests.

# QA steps
 - [x] automated tests passing

On local env, tested compatibility with,
 - [x] MongoDB v4.4
 - [x] MongoDB v.5.0
 - [x] MongoDB v6.0
 
# Risks

# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/test_mongodb_incremental_open_transactions.py [hunk 1]
```diff
@@ -306,3 +330,8 @@ def test_run(self):
 
             # validate that the state value has not changed after deleting the bookmarked value in each collection
             self.assertEqual(state_2, state_3)
+
+            # Reset the transaction life time limit
+            LOGGER.info(f"Re-setting transaction life time limit to {original_life_time_limit} seconds")
```
Comment (bhtowles) [inline tests/test_mongodb_incremental_open_transactions.py:335]: I've been informed that using f strings in LOGGER will fail pylint so I've been using:
LOGGER.info("Text %s, variable") format now

File: tests/test_mongodb_projection.py [hunk 2]
```diff
@@ -222,6 +222,10 @@ def run_single_projection(self, projection_mapping):
             #actual_keys = set()
 
             for record in stream_records:
+                # BUG TDL-23609. Pymongo v4.3+ return entire docuemtn for empty projection
```
Comment (bhtowles) [inline tests/test_mongodb_projection.py:225]: typo: document

File: tests/test_mongodb_incremental_open_transactions.py (hunk 1 above)
Comment (RushiT0122) [reply tests/test_mongodb_incremental_open_transactions.py:335]: I have not seen this issue. Also AFAIK pylist doesn't run on tests directory.

File: tests/test_mongodb_projection.py (hunk 2 above)
Comment (RushiT0122) [reply tests/test_mongodb_projection.py]: Fixed

File: tests/test_mongodb_incremental_open_transactions.py (hunk 1 above)
Comment (bhtowles) [reply tests/test_mongodb_incremental_open_transactions.py:335]: Correct, it only runs on ui-automation / webapp side.  But I try to be consistent across both

Comment (RushiT0122) [reply tests/test_mongodb_incremental_open_transactions.py]: Removed the `f-string` and replaced it with `.format()`.

File: Makefile [hunk 3]
```diff
@@ -1,4 +1,4 @@
 .DEFAULT_GOAL := test
 
 test:
-	pylint tap_mongodb tap_mongodb/sync_strategies -d missing-docstring,fixme,duplicate-code,line-too-long,too-many-statements,too-many-locals,consider-using-f-string,consider-using-from-import
+	pylint tap_mongodb tap_mongodb/sync_strategies -d missing-docstring,fixme,duplicate-code,line-too-long,too-many-statements,too-many-locals,consider-using-f-string,consider-using-from-import,,broad-exception-raised,superfluous-parens,consider-using-generator
```
Comment (luandy64) [inline Makefile:4]: Looks like there is an extra comma here. I'm a little shocked `pylint` doesn't complain about this 😂

File: tests/mongodb_common.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import os
 import pymongo
+from tap_tester.logger import  LOGGER
```
Comment (luandy64) [inline tests/mongodb_common.py:3]:
```suggestion
from tap_tester.logger import LOGGER
```

File: tests/test_mongodb_incremental_open_transactions.py [hunk 5]
```diff
@@ -306,3 +330,8 @@ def test_run(self):
 
             # validate that the state value has not changed after deleting the bookmarked value in each collection
             self.assertEqual(state_2, state_3)
+
+            # Reset the transaction life time limit
+            LOGGER.info("Re-setting transaction life time limit to {} seconds".format(original_life_time_limit))
+            success, _ = self.set_transaction_life_time_limit_seconds(client, original_life_time_limit)
+            self.assertTrue(success, msg="Failed to reset transaction life time limit")
```
Comment (luandy64) [inline tests/test_mongodb_incremental_open_transactions.py:337]: Does it make sense to fail the test if all test assertions pass, but the clean up fails?

Comment (RushiT0122) [reply tests/test_mongodb_incremental_open_transactions.py:337]: We are making changes to the database-level configuration. If we don't reset it then the remaining integration tests will run with the updated configuration, potentially causing us to miss critical issues.

**Review by luandy64 — APPROVED** [review]
Approving assuming the tests pass

**Pull request merged**